x/launch/client/cli/query_genesis_account.go

```diff
@@ -0,0 +1,82 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cast"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func CmdListGenesisAccount() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-genesis-account",
+		Short: "list all genesisAccount",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
+
+			params := &types.QueryAllGenesisAccountRequest{
+				Pagination: pageReq,
+			}
+
+			res, err := queryClient.GenesisAccountAll(context.Background(), params)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddPaginationFlagsToCmd(cmd, cmd.Use)
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdShowGenesisAccount() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show-genesis-account [chainID] [address]",
+		Short: "shows a genesisAccount",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+
+			queryClient := types.NewQueryClient(clientCtx)
+
+			argsChainID, err := cast.ToStringE(args[0])
```

Comment:
Why cast a string to string?

Reply:
Those are some cases where generated code is made a bit more complicated in order to simplify the template.

In this example, it simplifies the template because we can add a patter `cast.To<Type>E(...)` instead of having a if statement to handle different type.

But it's actually a good catch because we need this refactor this since it will not be a reliable solution when we will want to handle more advanced type for the field of a map